torch/_refs/__init__.py

```diff
@@ -1054,6 +1143,20 @@ def sub(
     aten_op=None,  # CompositeImplicitAutograd
 )
 
+
+def _trunc_divide(
+    a: Union[TensorLikeType, NumberType], b: Union[TensorLikeType, NumberType]
+):
+    return trunc(true_divide(a, b))
```

Comment:
I like how floor divide does so much work for numerical accuracy and then trunc division doesn't bother.

Did you happen to spot any examples where flooring the results of a true division is distinct from from the floor division calculation were performing? Are we perpetuating an implementation myth?

Reply:
Wait but with floor division shouldn't the difference have to be at least 1? Maybe we should show a specific input example in the comment so people understand why we're doing the algorithm.

Do you think we should implement a similar algorithm for trunc division, then?

Reply:
@mruberry I think we're following the `floor_divide` numpy behavior. Numpy does implement `trunc_divide`, so we implement `trunc_divide` as `trunc(div(a, b))`.

Example:
```Python
a = torch.tensor(-0.2500, dtype=torch.float64)
# tensor(-0.250000000000000, dtype=torch.float64)
#
b = torch.tensor(-0.0010, dtype=torch.float64)
# tensor(-0.001000000000000, dtype=torch.float64)
#
# Note: Casting float to double has different numerics from creating double directly
torch.tensor(-0.001).to(dtype=torch.float64)
# tensor(-0.001000000047497, dtype=torch.float64)
```

Floor Division
The difference is caused because `torch.remainder(a, b) = -0.001`.
```Python
torch.floor(torch.true_divide(a, b))
# tensor(250., dtype=torch.float64)
#
torch.div(a, b, rounding_mode='floor')
# tensor(249., dtype=torch.float64)
#
torch.true_divide(torch.sub(a, torch.remainder(a, b)), b)
# tensor(249., dtype=torch.float64)
```

Trunc Division
The difference is caused because `torch.fmod(a, b) = -0.001`.
```Python
torch.trunc(torch.true_divide(a, b))
# tensor(250., dtype=torch.float64)
#
torch.div(a, b, rounding_mode='trunc')
# tensor(250., dtype=torch.float64)
#
torch.true_divide(torch.sub(a, torch.fmod(a, b)), b)
# tensor(249., dtype=torch.float64)
```

Reply:
That's a helpful example. It's true that casting float to double is distinct from creating doubles directly, but we should be clear that only happens when the original value isn't exactly representable in fp32 and would have a distinct double representation from the float32 represenation

Reply:
Should this example be included with the eager mode definition or only for primtorch?